src/Content/Conversation/Repository/Channel.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2023, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Content\Conversation\Repository;
+
+use Friendica\BaseCollection;
+use Friendica\Content\Conversation\Entity\Timeline as TimelineEntity;
+use Friendica\Content\Conversation\Factory\Timeline;
+use Friendica\Database\Database;
+use Psr\Log\LoggerInterface;
+
+class Channel extends \Friendica\BaseRepository
+{
+	protected static $table_name = 'channel';
+
+	public function __construct(Database $database, LoggerInterface $logger, Timeline $factory)
+	{
+		parent::__construct($database, $logger, $factory);
+	}
+
+	/**
+	 * Fetch a single user channel
+	 *
+	 * @param int $id
+	 * @param int $uid
```

Comment:
Please mention this parameter is meant as a permission-checking mechanism since it isn't part of the primary key you're querying on.